src/project/environment.rs

```diff
@@ -35,6 +42,9 @@ pub struct Environment<'p> {
 
     /// The environment that this environment is based on.
     pub(super) environment: &'p manifest::Environment,
+
+    /// The cache that contains environment variables
+    vars: Arc<Mutex<HashMap<String, String>>>,
```

Comment:
This variable might be lost if someone requests the environment again from the project instead of cloning this type. It might be better to store a hashmap in the project, indexed by environment name.